### PR TITLE
Hide generator from settings

### PR DESCRIPTION
**Pull request by idoprz**
User can hide generators with a new setting

### CODE REVIEW

File: packages/backend/package.json [hunk 1]
```diff
@@ -99,6 +99,12 @@
           "description": "The verbosity of logging according to the following order: trace > debug > info > warn > error > fatal > off.",
           "scope": "resource"
         },
+        "ApplicationWizard.HideGenerator": {
+          "type": "string",
+          "default": "",
+          "description": "Hide the generator from the list of generators. Generator name should be in the format of namespace:generator-name. Generators should be seperated by comma.",
```
Comment (veredcon) [inline packages/backend/package.json:105]: seperated --> separated
Do you think Paola should review this string?

Comment (idoprz) [reply packages/backend/package.json:105]: does she review all strings of settings?

File: packages/backend/test/yeomanui.spec.ts [hunk 2]
```diff
@@ -545,12 +548,32 @@ describe("yeomanui unit test", () => {
...
         expect(result).to.be.deep.equal(noGeneratorsResult);
         expect(result.questions[0].choices).to.be.empty;
       });
 
-      it("generator should not be shown although there are additional categories'", async () => {
+      it("generator should not be shown because of settings", async () => {
+        const generatorData = {
+          generatorMeta: {
+            generatorPath: "test1Path/app/index.js",
+            namespace: "test1-project",
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:561]:
```suggestion
            namespace: "@ns/test1-project:app",
```

File: packages/backend/test/yeomanui.spec.ts [hunk 3]
```diff
@@ -545,12 +548,32 @@ describe("yeomanui unit test", () => {
...
+            packagePath: "test1Path",
+          },
+          generatorPackageJson: {
+            name: "@sap.koko",
+            description: "test hidden category",
+          },
+        };
+        wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("test1-project");
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:569]:
```suggestion
        wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("@ns/test1-project,other:app");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 4]
```diff
@@ -245,6 +245,7 @@ describe("yeomanui unit test", () => {
         });
       youiEventsMock.expects("setAppWizardHeaderTitle").withArgs(undefined);
       wsConfigMock.expects("get").withExactArgs("ApplicationWizard.TargetFolder").twice();
+      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").once();
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:248]:
```suggestion
      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").once().returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 5]
```diff
@@ -314,6 +315,7 @@ describe("yeomanui unit test", () => {
 
   it("_notifyGeneratorsChange", async () => {
     envUtilsMock.expects("getGeneratorsData").resolves();
+    wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").resolves();
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:318]: the test passes by chance - config.get() is a sync function that should return a string value (according to it signature. 
```suggestion
    wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 6]
```diff
@@ -487,6 +489,7 @@ describe("yeomanui unit test", () => {
 
     it("there are no generators", async () => {
       envUtilsMock.expects("getGeneratorsData").withExactArgs().resolves([]);
+      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").resolves();
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:492]:
```suggestion
      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 7]
```diff
@@ -545,12 +548,32 @@ describe("yeomanui unit test", () => {
           },
         };
         envUtilsMock.expects("getGeneratorsData").withExactArgs().resolves([generatorData]);
+        wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns({});
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:551]:
```suggestion
        wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 8]
```diff
@@ -559,6 +582,7 @@ describe("yeomanui unit test", () => {
...
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:585]:
```suggestion
        wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 9]
```diff
@@ -574,6 +598,7 @@ describe("yeomanui unit test", () => {
...
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:601]:
```suggestion
        wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 10]
```diff
@@ -584,6 +609,7 @@ describe("yeomanui unit test", () => {
 
     it("get generators with type project", async () => {
       wsConfigMock.expects("get").withExactArgs("ApplicationWizard.Workspace").returns({});
+      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").once();
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:612]:
```suggestion
      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").once().returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 11]
```diff
@@ -604,6 +630,7 @@ describe("yeomanui unit test", () => {
 
     it("get generators with type module", async () => {
       envUtilsMock.expects("getGeneratorsData").withExactArgs().resolves(gensMeta);
+      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").resolves();
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:633]:
```suggestion
      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 12]
```diff
@@ -616,6 +643,7 @@ describe("yeomanui unit test", () => {
 
     it("get generators all generators", async () => {
       envUtilsMock.expects("getGeneratorsData").resolves(gensMeta);
+      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").resolves();
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:646]:
```suggestion
      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 13]
```diff
@@ -627,6 +655,7 @@ describe("yeomanui unit test", () => {
 
     it("wrong generators filter type is provided", async () => {
       wsConfigMock.expects("get").withExactArgs("ApplicationWizard.Workspace").returns({});
+      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").resolves();
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:658]:
```suggestion
      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 14]
```diff
@@ -657,6 +686,7 @@ describe("yeomanui unit test", () => {
 
     it("get generators with type project and categories cat1 and cat2", async () => {
       wsConfigMock.expects("get").withExactArgs("ApplicationWizard.Workspace").returns({});
+      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns({});
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:689]:
```suggestion
      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 15]
```diff
@@ -697,6 +727,7 @@ describe("yeomanui unit test", () => {
         displayName: "3rd - Test",
       };
       envUtilsMock.expects("getGeneratorsData").withExactArgs().resolves(gensMeta.slice(0, 3));
+      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns({});
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:730]:
```suggestion
      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 16]
```diff
@@ -727,6 +758,7 @@ describe("yeomanui unit test", () => {
       };
 
       envUtilsMock.expects("getGeneratorsData").withExactArgs().resolves(gensMeta.slice(0, 3));
+      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns({});
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:761]:
```suggestion
      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("");
```

File: packages/backend/test/yeomanui.spec.ts [hunk 17]
```diff
@@ -769,7 +803,7 @@ describe("yeomanui unit test", () => {
       ]);
       wsConfigMock.expects("get").withExactArgs(yeomanUi["TARGET_FOLDER_CONFIG_PROP"]);
       wsConfigMock.expects("get").withExactArgs(yeomanUi["SELECTED_WORKSPACE_CONFIG_PROP"]);
-
+      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns({});
```
Comment (alex-gilin) [inline packages/backend/test/yeomanui.spec.ts:806]:
```suggestion
      wsConfigMock.expects("get").withExactArgs("ApplicationWizard.HideGenerator").returns("gen1,gen2,gen3");
```

**Review by alex-gilin — CHANGES_REQUESTED** [review]

**Review by idoprz — COMMENTED** [review]
I accepted all recommended changes

**Review by alex-gilin — APPROVED** [review]

**Pull request merged**